source/Posets/Delta-V-CompletePoset.lagda

```diff
@@ -0,0 +1,854 @@
+WEM, we now introduce the stronger concept of positivity.
+
+\begin{code}
+
+module Positive-Posets (𝓤  𝓦  𝓥 : Universe) (A : Poset 𝓤 𝓦) where
+ 
+ open δ-complete-poset 𝓥 A
+ open Universal fe
+ open PosetReasoning A
+ open Joins (_≤_)
+
+ module positive-posets (i : is-δ-complete) where
+
+  _<_ : (x y : ∣ A ∣ₚ) → 𝓤 ⊔ 𝓦 ⊔ (𝓥 ⁺) ̇ 
+  x < y = (x ≤ y) holds
+        × ((z : ∣ A ∣ₚ)
+          → (y ≤ z) holds
+          → (P : Ω 𝓥)
+          → (z is-lub-of (δ-fam x z P)) holds
+          → P holds)
+
+  order-< : {x y : ∣ A ∣ₚ} → x < y → (x ≤ y) holds
+  order-< c = pr₁ c
+
+  sup-condition : {x y : ∣ A ∣ₚ}
+                → x < y
+                → ((z : ∣ A ∣ₚ)
+                 → (y ≤ z) holds
+                 → (P : Ω 𝓥)
+                 → (z is-lub-of (δ-fam x z P)) holds
+                 → P holds)
+  sup-condition c = pr₂ c
+
+  strictly-below-implies-non-trivial : (x y : ∣ A ∣ₚ)
+                                     → is-δ-complete
+                                     → (x < y)
+                                     → (x ≤ y) holds × (x ≠ y)
+  pr₁ (strictly-below-implies-non-trivial x y i c) = order-< c
+  pr₂ (strictly-below-implies-non-trivial x y i c) p =
+   𝟘-induction (sup-condition c y (≤-is-reflexive A y)
+               ⊥
+               (y-is-ub , y-has-lub-cond))
+    where
+     y-is-ub : (y is-an-upper-bound-of (δ-fam x y ⊥)) holds
+     y-is-ub (inl ⋆) = order-< c
+
+     y-has-lub-cond : (u : upper-bound (δ-fam x y ⊥)) → (y ≤ (pr₁ u)) holds
+     y-has-lub-cond u = y ＝⟨ p ⁻¹ ⟩ₚ pr₂ u (inl ⋆)
+
+\end{code}
+
+We could show that if the converse holds then so does EM in 𝓥.
+
+\begin{code}
+
+  ≤-<-to-< : (i : is-δ-complete)
+           → (x y z : ∣ A ∣ₚ)
+           → (x ≤ y) holds × y < z
+           → x < z 
+  ≤-<-to-< i x y z (x-≤-y , y-<-z) =
+    (≤-is-transitive A x y z x-≤-y (order-< y-<-z) , sup-cond-P)
+   where
+    sup-cond-P : (w : ∣ A ∣ₚ)
+               → (z ≤ w) holds
+               → (P : Ω 𝓥)
+               → (w is-lub-of (δ-fam x w P)) holds
+               → P holds
+    sup-cond-P w z-≤-w P (w-is-ubₓ , w-has-lub-condₓ) =
+      sup-condition y-<-z w z-≤-w P (w-is-ub , w-has-lub-cond)
+     where
+      w-is-ub : (w is-an-upper-bound-of (δ-fam y w P)) holds
+      w-is-ub (inl ⋆) = ≤-is-transitive A y z w (order-< y-<-z) z-≤-w
+      w-is-ub (inr p) = ≤-is-reflexive A w
+
+      w-has-lub-cond : ((u , u-is-ub) : (upper-bound (δ-fam y w P)))
+                     → (w ≤ u) holds 
+      w-has-lub-cond (u , u-is-ub) = w-has-lub-condₓ (u , u-is-ubₓ)
+       where
+        u-is-ubₓ : (u is-an-upper-bound-of (δ-fam x w P)) holds
+        u-is-ubₓ (inl ⋆) = ≤-is-transitive A x y u (x-≤-y) (u-is-ub (inl ⋆))
+        u-is-ubₓ (inr p) = u-is-ub (inr p)
+
+  <-≤-to-< : (i : is-δ-complete)
+           → (x y z : ∣ A ∣ₚ)
+           → x < y × (y ≤ z) holds
+           → x < z
+  <-≤-to-< i x y z (x-<-y , y-≤-z) =
+   (≤-is-transitive A x y z (order-< x-<-y) y-≤-z , sup-cond-P)
```

Comment:
Here are some other examples of the discouraged naming scheme: `x-<-y`, `y-≤-z`.